src/main/java/com/commercetools/project/sync/service/impl/ReferencesServiceImpl.java

```diff
@@ -24,36 +34,40 @@ public ReferencesServiceImpl(@Nonnull final SphereClient ctpClient) {
   }
 
   /**
-   * Given 3 {@link Set}s of ids of products, categories and productTypes, this method first checks
-   * if there is a key mapping for each id in the {@code idToKey} cache. If there exists a mapping
-   * for all the ids, the method returns a future containing the existing {@code idToKey} cache as
-   * it is. If there is at least one missing mapping, it attempts to make a GraphQL request to CTP
-   * to fetch all ids and keys of every missing product, category or productType Id in a combined
-   * request. For each fetched key/id pair, the method will insert it into the {@code idToKey} cache
-   * and then return the cache in a {@link CompletableFuture} after the request is successful.
+   * Given 4 {@link Set}s of ids of products, categories, productTypes and custom objects, this
+   * method first checks if there is a key mapping for each id in the {@code idToKey} cache. If
+   * there exists a mapping for all the ids, the method returns a future containing the existing
+   * {@code idToKey} cache as it is. If there is at least one missing mapping, it attempts to make a
+   * GraphQL request (note: rest request for custom objects) to CTP to fetch all ids and keys of
+   * every missing product, category, productType or custom object Id in a combined request. For
+   * each fetched key/id pair, the method will insert it into the {@code idToKey} cache and then
+   * return the cache in a {@link CompletableFuture} after the request is successful.
    *
    * @param productIds the product ids to find a key mapping for.
    * @param categoryIds the category ids to find a key mapping for.
    * @param productTypeIds the productType ids to find a key mapping for.
-   * @return a map of id to key representing products, categories and productTypes in the CTP
-   *     project defined by the injected {@code ctpClient}.
+   * @param customObjectIds the custom object ids to find a key mapping for.
+   * @return a map of id to key representing products, categories, productTypes and customObjects in
+   *     the CTP project defined by the injected {@code ctpClient}.
    */
   @Nonnull
   @Override
   public CompletionStage<Map<String, String>> getIdToKeys(
       @Nonnull final Set<String> productIds,
       @Nonnull final Set<String> categoryIds,
-      @Nonnull final Set<String> productTypeIds) {
+      @Nonnull final Set<String> productTypeIds,
+      @Nonnull final Set<String> customObjectIds) {
 
     final Set<String> nonCachedProductIds = getNonCachedIds(productIds);
     final Set<String> nonCachedCategoryIds = getNonCachedIds(categoryIds);
     final Set<String> nonCachedProductTypeIds = getNonCachedIds(productTypeIds);
+    final Set<String> nonCachedCustomObjectIds = getNonCachedIds(customObjectIds);
 
     // if everything is cached, no need to make a request to CTP.
```

Comment:
custom object keys are not cached yet. So, we cannot say `if everything is cached` anymore. Shouldn't we change the comment slightly?

Reply:
https://github.com/commercetools/commercetools-project-sync/pull/176/commits/3bc33b61b25894111d0ef02ec8acb11ec957e1f7